### PR TITLE
ActionObjects tests and docs

### PR DESCRIPTION
**Pull request by koenvanderveen**
## Description
basic eager execution functionality:
 - [x] Improve ActionObject error handling
 - [x] Improve ActionObject docstrings
 - [x] Test coverage for basic operations for types: int, float, list, set, dict, tuple.
 - [x] Complete dunder methods. 
 - [x] Bugfixing: LineageID hashing errors. LineageID vs UID comparison errors.
 - [x] Automatic args/kwargs pointerization
 - [x] set/get attribute support.

Leftoevers:
 - The Hooks are not handling errors.
 - `dict_keys` is not supported in ActionObject
 - 
related to https://github.com/OpenMined/Heartbeat/issues/191
fixes https://github.com/OpenMined/PySyft/issues/7500

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: packages/syft/src/syft/core/node/new/action_object.py [hunk 1]
```diff
@@ -321,57 +370,247 @@ def syft_is_property(self, obj: Any, method: str) -> bool:
...
+        """Get the object from a Syft Client"""
+
         return client.api.services.action.get(self.id).syft_action_data
 
     @staticmethod
     def from_obj(
         syft_action_data: Any,
         id: Optional[UID] = None,
         syft_lineage_id: Optional[LineageID] = None,
-    ) -> ActionObject:
```
Comment (koenvanderveen) [inline packages/syft/src/syft/core/node/new/action_object.py:335]: personally I am not sold on the `ActionObject.from_obj(x).ok()` dev experience. Curious to hear what others thinks

File: packages/syft/src/syft/core/node/new/action_object.py [hunk 2]
```diff
@@ -36,6 +39,23 @@
 
 @serializable()
 class Action(SyftObject):
+    """Serializable Action object.
+
+    Parameters:
+        path: str
+            The Type of the remote object.
```
Comment (koenvanderveen) [inline packages/syft/src/syft/core/node/new/action_object.py:46]: is this actually the type? or is it the path to the type?

File: packages/syft/src/syft/core/node/new/action_object.py [hunk 3]
```diff
@@ -135,110 +160,137 @@ class ActionObjectPointer:
...
+        obj: Any
+            The ActionObject to use for the action
+        op_name: str
+            The method name to use for the action
+        node_uid: Optional[UID]
+            Optional Syft node UID
+        result_id: Optional[Union[UID, LineageID]]
+            Optional result Syft UID
+        action: Optional[Action]
+            The action generated by the hook
```
Comment (koenvanderveen) [inline packages/syft/src/syft/core/node/new/action_object.py:178]: Is this the current hook, or previous hooks?

File: packages/syft/src/syft/core/node/new/action_object.py (hunk 1 above)
Comment (koenvanderveen) [reply packages/syft/src/syft/core/node/new/action_object.py:335]: not sure if users would use it directly though, or always work with something that is generating those

File: packages/syft/tests/syft/actions/action_object_test.py [hunk 4]
```diff
@@ -0,0 +1,993 @@
...
+        (True, "__and__", [False], {}),
+        ((1, 1, 3), "count", [1], {}),
+        ([1, 2, 1], "count", [1], {}),
+        ([1, 2, 3], "append", [4], {}),
+        ({"a": 1, "b": 2}, "update", [{"c": 3}], {}),
+        (set({1, 2, 3, 3}), "add", [5], {}),
+        (set({1, 2, 3, 3}), "clear", [], {}),
+    ],
+)
+def test_actionobject_syft_make_remote_method_action(worker, testcase):
```
Comment (koenvanderveen) [inline packages/syft/tests/syft/actions/action_object_test.py:479]: I dont think this method is used anywhere, maybe we should remove it. cc @madhavajay

**Comment by koenvanderveen** [comment]
This looks great to me! The added test coverage is awesome and I really like the docs.

File: packages/syft/tests/syft/actions/action_service_test.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+# syft absolute
+from syft.core.node.new.action_object import ActionObject
+from syft.core.node.new.context import AuthedServiceContext
+
+
+def get_auth_ctx(worker):
+    return AuthedServiceContext(node=worker, credentials=worker.signing_key.verify_key)
+
+
+def test_action_service_sanity(worker):
```
[the remark this answers is not in the document]
Comment (bcebere) [reply packages/syft/tests/syft/actions/action_service_test.py:14]: This PR is only about ActionObject, I will have a look at the ActionService in another PR.

File: packages/syft/src/syft/core/node/new/action_object.py (hunk 1 above)
Comment (bcebere) [reply packages/syft/src/syft/core/node/new/action_object.py:335]: Sure, but at some point, the library should stick to some error handling strategy. And Result seems to be already used a lot in the library.

**Review by madhavajay — APPROVED** [review]
Great work documenting these methods and cleaning up the hook code.

**Pull request merged**